app/screens/settings/notification_push/push_status.tsx

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {View} from 'react-native';
+
+import Block from '@components/block';
+import BlockItem from '@components/block_item';
+import FormattedText from '@components/formatted_text';
+import {useTheme} from '@context/theme';
+import {t} from '@i18n';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+const headerText = {
+    id: t('notification_settings.mobile.push_status'),
+    defaultMessage: 'Trigger push notifications when',
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        upperCase: {
+            textTransform: 'uppercase',
+        },
+        separator: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.1),
+            flex: 1,
+            height: 1,
+            marginLeft: 15,
+        },
+        area: {
+            paddingHorizontal: 16,
+        },
+        label: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100, 'Regular'),
+
+        },
+    };
+});
+
+type MobilePushStatusProps = {
+    sendPushNotifications: boolean;
+    pushStatus: PushStatus;
+    setMobilePushStatus: (status: PushStatus) => void;
+}
+const MobilePushStatus = ({sendPushNotifications, pushStatus, setMobilePushStatus}: MobilePushStatusProps) => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+    const showSection = sendPushNotifications && pushStatus !== 'none';
```

Comment:
I think you know this before the component is mounted, that way you can decide to mount or not the component instead of rendering null, am I right?

Reply:
corrected